dependencies/src/test/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphIntegrationTest.java

```diff
@@ -0,0 +1,145 @@
+import org.eclipse.aether.resolution.DependencyResolutionException;
+import org.junit.Assert;
+import org.junit.Test;
+
+import com.google.common.truth.Truth;
+
+/**
+ *  Tests against actual artifacts in Maven central.
+ */
+public class DependencyGraphIntegrationTest {
+
+  @Test
+  public void testFindUpdates()
+      throws DependencyCollectionException, DependencyResolutionException {
+    
+    DefaultArtifact core =
+        new DefaultArtifact("com.google.cloud:google-cloud-core:1.37.1");
+    
+    DependencyGraph graph = DependencyGraphBuilder.getCompleteDependencies(core);
+    List<Update> updates = graph.findUpdates();
+    List<String> strings = updates.stream().map(e -> e.toString()).collect(Collectors.toList());
+    
+    // ordering not working yet
+    // TODO get order working
+    Truth.assertThat(strings).containsExactly("com.google.guava:guava:20.0 needs to "
+        + "upgrade com.google.code.findbugs:jsr305:1.3.9 to 3.0.2",
+        "com.google.http-client:google-http-client:1.23.0 needs to "
+        + "upgrade com.google.code.findbugs:jsr305:1.3.9 to 3.0.2",
+        "com.google.api:api-common:1.6.0 needs to "
+        + "upgrade com.google.code.findbugs:jsr305:3.0.0 to 3.0.2",
+        "com.google.api.grpc:proto-google-common-protos:1.12.0 needs to "
+        + "upgrade com.google.protobuf:protobuf-java:3.5.1 to 3.6.0",
+        "com.google.api.grpc:proto-google-iam-v1:0.12.0 needs to "
+        + "upgrade com.google.protobuf:protobuf-java:3.5.1 to 3.6.0",
+        "com.google.api.grpc:proto-google-iam-v1:0.12.0 needs to "
+        + "upgrade com.google.api.grpc:proto-google-common-protos:1.11.0 to 1.12.0",
+        "com.google.api.grpc:proto-google-iam-v1:0.12.0 needs to "
+        + "upgrade com.google.api:api-common:1.5.0 to 1.6.0",
+        "com.google.api:api-common:1.6.0 needs to "
+        + "upgrade com.google.guava:guava:19.0 to 20.0",
+        "com.google.auth:google-auth-library-oauth2-http:0.9.1 needs to "
+        + "upgrade com.google.guava:guava:19.0 to 20.0",
+        "com.google.protobuf:protobuf-java-util:3.6.0 needs to "
+        + "upgrade com.google.guava:guava:19.0 to 20.0",
+        "com.google.auth:google-auth-library-oauth2-http:0.9.1 needs to "
+        + "upgrade com.google.http-client:google-http-client:1.19.0 to 1.23.0",
+        "com.google.http-client:google-http-client-jackson2:1.19.0 needs to "
+        + "upgrade com.google.http-client:google-http-client:1.19.0 to 1.23.0");
+  }
+  
+  @Test
+  // beam has a more complex dependency graph that hits some corner cases
+  public void testFindUpdates_beam()
+      throws DependencyCollectionException, DependencyResolutionException {    
+
+    DefaultArtifact beam =
+        new DefaultArtifact("org.apache.beam:beam-sdks-java-io-google-cloud-platform:2.5.0");
+    DependencyGraph graph = DependencyGraphBuilder.getCompleteDependencies(beam);
+    
+    graph.findUpdates();
```

Comment:
Yep, nothing is thrown. This test is here because it used to fail. Only moved in this PR. In particular there are some weirdnesses in the Beam dependency graph where resolution of the correct artifact requires detecting which operating system you're running on.

Reply:
Can you add a comment to this effect here?

Reply:
done